apps/experiments/migrations/0084_participant_platform.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 4.2.11 on 2024-07-17 13:49
+
+from django.db import migrations, models
+import django.db.models.deletion
+from django.db.models import OuterRef, Subquery, Value
+from django.db.models.functions import Coalesce
+
+
+def set_participant_platform(apps, schema_editor):
+    Participant = apps.get_model("experiments", "Participant")
+    ExperimentSession = apps.get_model("experiments", "ExperimentSession")
+    Participant.objects.update(
+        platform=Coalesce(
```

Comment:
read up on the docs for Coalesce and Subquery since I haven't seen that syntax before, what was the give away that this needed to be used rather than a basis query?

Reply:
Also a question from my side out of curiosity: Did you encounter a case where `experiment_channel__platform` is `None`, or is the usage of `Coalesce` just a safety layer for incase?

Reply:
@stephherbers did you mean `basic` query (I'm not sure what a `basis` query is).

They subquery is required since I'm updating the column with a value from a different table so I need to do a join. In PostgreSQL you can do that with `update .. from` syntax but Django only supports doing it with a subquery.

The `Coalesce` was needed because `session.experiment_channel` is an optional field (if you delete a channel it becomes `null`).

Reply:
Oh I see, since we exclude sessions where the channel is None, it could be that there are none left..thanks!

Reply:
haha no basis was a typo, thanks for the explaination